merkle_zeppelin/trees/merkle.py

```diff
@@ -16,67 +20,141 @@ def keccak256(v: bytes) -> bytes:
 class MerkleTree(BinaryTree[bytes]):
     def __init__(
         self,
-        leaves: list[Leaf],
+        raw_elements: list[Leaf],
         types: list[str],
         hashing_function: Callable[[bytes], bytes] = None,
     ) -> None:
         self._hashing_function = hashing_function or keccak256
         self._types = types
 
-        hash_leaf_pairs = self._get_hash_leaf_pairs(leaves)
-        hash_leaf_pairs.sort(reverse=True)
+        sorted_index_hash_pairs = self._get_sorted_index_hash_pairs(raw_elements)
+        sorted_hashes = [hash_ for index, hash_ in sorted_index_hash_pairs]
+
+        super().__init__(sorted_hashes)
 
-        ordered_hashed_leaves = [leaf_hash for leaf_hash, _ in hash_leaf_pairs]
-        ordered_leaves = [leaf for _, leaf in hash_leaf_pairs]
+        self._raw_to_leaves_index_mapping = self._get_raw_to_leaves_index_mapping(
+            raw_elements, sorted_index_hash_pairs
+        )
 
-        super().__init__(ordered_hashed_leaves)
+    @staticmethod
+    def get_hash_from_string(hash_: str) -> bytes:
+        if hash_.startswith("0x"):
+            hash_ = hash_[2:]
 
-        self._raw_leaves_index = self._get_raw_leaves_index(ordered_leaves)
+        return bytes.fromhex(hash_)
+
+    @classmethod
+    def import_tree(
+        cls,
+        data: Any,
+        importer: Type[MerkleTreeImporter],
+        validate: bool = True,
+        hashing_function: Callable[[bytes], bytes] = None,
+    ) -> MerkleTree:
+        imported_data = importer.import_tree(data)
+        obj = cls.__new__(cls)
+        obj._hashing_function = hashing_function or keccak256
+        obj._nodes = [
+            cls.get_hash_from_string(hash_str) for hash_str in imported_data.tree
+        ]
+        obj._types = imported_data.leaf_encoding
+        obj._raw_to_leaves_index_mapping = {
+            value.value: value.tree_index for value in imported_data.values
+        }
```

Comment:
Optional suggestion:
Maybe I'm missing something here, but IMO since the return type of `importer.import_tree` is always `MerkleTreeDTO` (or descedants), I would use `__init__()` method here, i.e. `cls()`, or implement some special constructor to load the `MerkleTreeDTO` object, i.e. `cls.load_from_dto(imported_data)`

Reply:
Forgot to add "why" :)
This feels a bit like breaking your own rules when it is unnecessary and can be pretty easy to forget to update when you change the `__init__` method of the `MerkleTree` (e.g. adding/removing some extra data). 

I'd try to make this method work independently from changes in the `MerkleTree` and `MerkleTreeDTO` structure.

Reply:
huh, it took me awhile but finally I got you!
tough decision
in your solution class needs two paths in `__init__()` - one of them is path with allowed empty / None arguments provided, which in my opinion should be not allowed,
but on the other hand - you have valid point about possibility to forget to update both places `__init__` and import function

I will hold my decision and see if there is some third option

Reply:
> in your solution class needs two paths in __init__() - one of them is path with allowed empty / None arguments provided, which in my opinion should be not allowed,

Not really. As I also mentioned, you could implement a separate class method to load from dto object. But again, this is up to you.

Reply:
ok, now we have separate function for importing from DTO, it is good idea : )